File: chatbot-directline-api/python_demo/step2_start_conversation.py
import requests
import os


def start_conversation():
    """Step 2: Start a new conversation session"""
    print("Step 2: Starting conversation...")

    # Get the directory where this script is located
    script_dir = os.path.dirname(os.path.abspath(__file__))
    data_dir = os.path.join(script_dir, "data")
    os.makedirs(data_dir, exist_ok=True)

    # Read token from previous step
    try:
        with open(os.path.join(data_dir, "token.txt"), "r") as f:
            token = f.read().strip()
    except FileNotFoundError:
        print("❌ Token not found. Run step1_generate_token.py first.")
        return None

    url = "https://directline.botframework.com/v3/directline/conversations"
    headers = {
        "Authorization": f"Bearer {token}",
        "Content-Type": "application/json",
    }

    # Send request to start conversation
    response = requests.post(url, headers=headers, json={})

    if response.status_code in [200, 201]:
        data = response.json()
        conversation_id = data.get("conversationId")
        updated_token = data.get("token")

        print(f"✅ Conversation started successfully!")
        print(f"   Conversation ID: {conversation_id}")
        print(f"   Stream URL available: {'Yes' if data.get('streamUrl') else 'No'}")

        # Save updated token and conversation ID
        with open(os.path.join(data_dir, "token.txt"), "w") as f:
            f.write(updated_token)
        with open(os.path.join(data_dir, "conversation_id.txt"), "w") as f:
            f.write(conversation_id)

        return conversation_id
    else:
        print(f"❌ Failed to start conversation: {response.text}")
        return None


if __name__ == "__main__":
    start_conversation()

File: chatbot-directline-api/php_demo/step2_start_conversation.php
<?php
/**
 * Step 2: Start a new conversation session
 */

require_once 'config.php';

function startConversation() {
    echo "Step 2: Starting conversation...\n";
    
    // Ensure data directory exists
    $dataDir = 'data';
    if (!is_dir($dataDir)) {
        mkdir($dataDir, 0755, true);
    }
    
    // Read token from previous step
    $tokenFile = $dataDir . '/token.txt';
    if (!file_exists($tokenFile)) {
        echo "❌ Token not found. Run step1_generate_token.php first.\n";
        return null;
    }
    
    $token = trim(file_get_contents($tokenFile));
    
    $url = 'https://directline.botframework.com/v3/directline/conversations';
    $headers = [
        'Authorization: Bearer ' . $token,
        'Content-Type: application/json'
    ];
    
    // Prepare cURL request
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_POST, true);
    curl_setopt($ch, CURLOPT_POSTFIELDS, '{}');
    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, true);
    
    // Send request
    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);
    
    if ($httpCode === 200 || $httpCode === 201) {
        $data = json_decode($response, true);
        $conversationId = $data['conversationId'] ?? null;
        $updatedToken = $data['token'] ?? null;
        
        if ($conversationId) {
            echo "✅ Conversation started successfully!\n";
            echo "   Conversation ID: {$conversationId}\n";
            echo "   Stream URL available: " . (isset($data['streamUrl']) ? 'Yes' : 'No') . "\n";
            
            // Save updated token and conversation ID
            if ($updatedToken) {
                file_put_contents($dataDir . '/token.txt', $updatedToken);
            }
            file_put_contents($dataDir . '/conversation_id.txt', $conversationId);
            
            return $conversationId;
        }
    }
    
    echo "❌ Failed to start conversation: HTTP {$httpCode}\n";
    if ($response) {
        echo "Response: {$response}\n";
    }
    
    return null;
}

// Run if called directly
if (basename(__FILE__) === basename($_SERVER['SCRIPT_NAME'])) {
    startConversation();
}
?> 